### PR TITLE
Add bonds

### PR DESCRIPTION
**Pull request by siegfault**
Treated as a type of item, much like armor, spell, or class, bonds link characters in up to three ways to other entities. The source material does not give a name to each of the three axes that make up a bond, so I've chosen a name for each as follows.

```
Admiration vs. Inferiority => Respect
Loyalty vs. Mistrust => Reliance
Affection vs. Hatred => Disposition
```

On each axis, the value is stored as a string - either an empty string to represent no preference, or the positive or negative affect.

### CODE REVIEW

File: src/actor/character/sheet.js [hunk 1]
```diff
@@ -45,12 +46,100 @@ export class FabulaUltimaCharacterSheet extends FabulaUltimaActorSheet {
...
+      if (i.type === 'bond') {
+        bonds.push(i);
+      }
+    }
+
+    // Assign and return
+    context.bonds = bonds;
+  }
+  /** override **/
+  activateListeners(html) {
```
Comment (AllPurposeName) [inline src/actor/character/sheet.js:102]: I am just going to nod and say "sounds good" to this entire function.

File: src/actor/character/sheets-tabs/main-tab.hbs [hunk 2]
```diff
@@ -111,54 +111,7 @@
...
-            <p> <input type="checkbox" name="loyalty" /> Loyalty</p>
-            <p> <input type="checkbox" name="mistrust" /> Mistrust</p>
-          </div>
-          <div class="bond-feels">
-            <p> <input type="checkbox" name="affection" /> Affection</p>
-            <p> <input type="checkbox" name="hatred" /> Hatred</p>
-          </div>
-        </div>
-      </div>
+      {{> "systems/fabula-ultima/templates/actor/character/parts/bonds.hbs" }}
```
Comment (AllPurposeName) [inline src/actor/character/sheets-tabs/main-tab.hbs:114]: Ooh I like this

File: src/system/core/config.js [hunk 3]
```diff
@@ -150,6 +150,7 @@ FabulaUltima.i18n = {
 FabulaUltima.itemTypes = [
         "armor",
         "artifact",
+  "bond",
         "helper",
```
Comment (AllPurposeName) [inline src/system/core/config.js:154]:
```suggestion
        "artifact",
        "bond",
        "helper",
```

File: src/system/core/handlebars.js [hunk 4]
```diff
@@ -193,6 +193,20 @@ function registerHandlebarsHelpers() {
     return new Handlebars.SafeString(value.replace(/(<([^>]+)>)/gi, "").replace(/(?:\r\n|\r|\n)/g, "<br/>"));
   });
 
+  Handlebars.registerHelper("feelingChoice", function (current_selection, axis, choice) {
```
Comment (AllPurposeName) [inline src/system/core/handlebars.js:196]: I really like how this turned out.

File: src/system/core/handlebars.js [hunk 5]
```diff
@@ -193,6 +193,20 @@ function registerHandlebarsHelpers() {
...
+    var uppercased = Handlebars.helpers.toUpperCase(choice);
+    var translation = game.i18n.localize(`BOND.${uppercased}`);
+    var is_checked = Handlebars.helpers.checked(current_selection === choice);
+
+    return `
+      <div class="flexrow bond-feels">
+        <input type="checkbox" name="${choice}" class="feeling-checkbox" data-dtype="Boolean" ${is_checked} data-prop="${axis}" />
+        <label for="feelings.${choice}" class="resource-label feeling flexlarge align-left" data-label="${translation}">${translation}</l>
+
```
Comment (AllPurposeName) [inline src/system/core/handlebars.js:205]:
```suggestion
```

**Review by AllPurposeName — APPROVED** [review]
A couple non-blocking suggestions, but it looks great!

File: src/actor/character/sheet.js (hunk 1 above)
Comment (siegfault) [reply src/actor/character/sheet.js:102]: FWIW, I think it was more or less modeled after https://github.com/fvtt-fria-ligan/forbidden-lands-foundry-vtt/blob/78c390afcc7eae5683e608a75578be4ed04ddb19/src/actor/character/character-sheet.js#L53

File: src/system/core/config.js (hunk 3 above)
Comment (siegfault) [reply src/system/core/config.js:154]: I'm fine changing this to be in line. We should probably either keep tabs though or convert the entire codebase to spaces at some point

File: src/system/core/handlebars.js (hunk 4 above)
Comment (siegfault) [reply src/system/core/handlebars.js:196]: Thanks! If we had access to bond within the helper, I think we could've gotten away without including the first argument, but oh well

**Pull request merged**